pkg/kubectl/cmd/util/helpers.go

```diff
@@ -243,6 +243,23 @@ func MultilineError(prefix string, err error) string {
 	return fmt.Sprintf("%s%s\n", prefix, err)
 }
 
+// PrintErrorWithCauses prints an error's kind, name, and each of the error's causes in a new line.
+// The returned string will end with a newline.
+// Returns true if a case exists to handle the error type, or false otherwise.
+func PrintErrorWithCauses(err error, errOut io.Writer) bool {
+	switch t := err.(type) {
+	case *kerrors.StatusError:
+		errorDetails := t.Status().Details
+		fmt.Fprintf(errOut, "error: %s %q is invalid\n\n", errorDetails.Kind, errorDetails.Name)
```

Comment:
@janetkuo Thanks for the feedback, I've updated this to handle `errorDetails` being `nil`.

> Why two newlines?

This is to separate the `error: ... is invalid` line from the detailed error causes:
```
$ kubectl set probe resourecekind/myresource --liveness --get-url=www.google.com:80
error: ResourceKind "myResource" is invalid

* spec.template.spec.containers[0].livenessProbe.httpGet.port: Invalid value: "": must contain only alpha-numeric characters (a-z, 0-9), and hyphens (-)
* spec.template.spec.containers[0].livenessProbe.httpGet.port: Invalid value: "": must contain at least one letter (a-z)
* spec.template.spec.containers[0].livenessProbe.httpGet.scheme: Unsupported value: "WWW.GOOGLE.COM": supported values: HTTP, HTTPS
```